src/mmw/js/src/analyze/models.js

```diff
@@ -54,27 +54,37 @@ var AnalyzeTaskModel = coreModels.TaskModel.extend({
     }
 });
 
-AnalyzeTaskModel.getSingleton = function(App, aoi) {
-    if (!App.analyzeModel) {
-        App.analyzeModel = createTaskModel(JSON.stringify(aoi));
+var AnalyzeTaskCollection = Backbone.Collection.extend({
+    model: AnalyzeTaskModel
+});
+
+AnalyzeTaskCollection.getSingleton = function(App, aoi) {
+    if (!App.analyzeCollection) {
+        App.analyzeCollection = new AnalyzeTaskCollection([
+            { area_of_interest: aoi, taskName: 'analyze' },
+            { area_of_interest: aoi, taskName: 'analyze/animals' },
+            { area_of_interest: aoi, taskName: 'analyze/pointsource' },
+        ]);
     }
 
-    return App.analyzeModel;
+    return App.analyzeCollection;
 };
 
-// Pass in the serialized Area of Interest for
-// caching purposes (_.memoize returns the same
-// results for any object), and deserialize
-// the AoI for use on the model.
-var createTaskModel = _.memoize(function(aoi) {
-    return new AnalyzeTaskModel({
-        area_of_interest: JSON.parse(aoi)
-    });
-});
+function createAnalyzeResultViewModelCollection(App, aoi) {
```

Comment:
I think I would keep this logic here, but I would move the actual singleton getter/setter logic to `App`.

The rationale for this is to encapsulate implementation logic inside `App` and avoid mutating it from an external function.

So instead of

```
var App = { };

function getSingleton(App) {
    if (!App.analyzeCollection) {
        App.analyzeCollection = { ... };
    }
    return App.analyzeCollection;
}

getSingleton(App);
delete App.analyzeCollection;
```

We could do

```
var App = {
    getAnalyzeCollection: function() {
        if (!this.analyzeCollection) {
            this.analyzeCollection = createAnalyzeCollection();
        }
        return this.analyzeCollection;
    },
    clearAnalyzeCollection: function() {
        delete this.analyzeCollection;
    }
};

function createAnalyzeCollection() {
    return { ... };
}

App.getAnalyzeCollection();
App.clearAnalyzeCollection();
```